Content.Shared/Weapons/Melee/SharedMeleeWeaponSystem.cs

```diff
@@ -360,27 +360,28 @@ public float GetModifier(MeleeWeaponComponent component, bool lightAttack)
 
     protected virtual void DoLightAttack(EntityUid user, LightAttackEvent ev, MeleeWeaponComponent component, ICommonSession? session)
     {
+        var damage = component.Damage * GetModifier(component, true);
+
         // Can't attack yourself
         // Not in LOS.
         if (user == ev.Target ||
             ev.Target == null ||
             Deleted(ev.Target) ||
             // For consistency with wide attacks stuff needs damageable.
             !HasComp<DamageableComponent>(ev.Target) ||
-            !TryComp<TransformComponent>(ev.Target, out var targetXform))
+            !TryComp<TransformComponent>(ev.Target, out var targetXform) ||
+            !InRange(user, ev.Target.Value, component.Range, session))
         {
-            Audio.PlayPredicted(component.SwingSound, component.Owner, user);
-            return;
-        }
+            // Leave IsHit set to true, because the only time it's set to false
+            // is when a melee weapon is examined. Misses are inferred from an
+            // empty HitEntities.
+            var missEvent = new MeleeHitEvent(new List<EntityUid>(), user, damage);
+            RaiseLocalEvent(component.Owner, missEvent);
```

Comment:
Adding this to the same if is changing the behaviour; if you click yourself or clicked a deleted target it shouldn't be doing any form of attack.

Reply:
I've reorganized the conditions. How does it look now?